pytest_testrail/plugin.py

```diff
@@ -153,7 +155,18 @@ def pytest_collection_modifyitems(self, session, config, items):
         tr_keys = [case_id for item in items_with_tr_keys for case_id in item[1]]
 
         if self.testplan_id and self.is_testplan_available():
-            self.testrun_id = 0
+            if self.testrun_name is None:
+                self.testrun_name = testrun_name()
+
+            self.add_plan_entry(
```

Comment:
If I understand your code, we can't no more manage the following case: Publish test results among all testruns contained in a testplan. However, it's for this original case that `--tr-plan-id` option has been created.

See [this code](https://github.com/allankp/pytest-testrail/blob/92d0a279da0f12b912c3ad5c65561a3fb8c874a7/pytest_testrail/plugin.py#L206) for a better understanding.

I think you should check that no testrun exists before creating a new.

Reply:
Sending results to a known run in a test plan makes perfect sense to me, but I can't imagine a use case for sending results to all runs in a test plan, at least not the way we're using test plans, to represent different browsers, environments, etc. For us, any physical run would always be appropriate to record results in just just one Test Rail test run... Am I missing a use case that folks are using? If I check that to see run exists, but I find 1 or more, I should go ahead and use one or more to capture results, even though no run id was passed in? If I find 1 or more, I should go ahead and use the unbounded (no test run id) set and record results on all of them? Makes no sense to me. In our case, we want to accumulate runs in the test plan, so it is expected that others will exist and will not be touched by the new run.

Reply:
Looking at the test for `create_test_run`, and trying to use it as a model for tests against `add_plan_entry`, I have to admit I'm over my head. I'll give this as much time as I can, but if getting unit tests accepted along with the rest of PR is required, I guess I'll have to run off my fork...

Reply:
@conversica-aaronpa I've not had time to take a look at this yet, I will however take some time on Friday to run through this and help with unit tests as these are required to be implemented and passing before getting merged.

Reply:
Tests are passing, I'm just struggling to understand and being able to assert that they do appropriate validations.

Reply:
@conversica-aaronpa Sorry if the actual behavior doesn't fit your needs. The use case you spoke ("sending results to all runs in a test plan") is the one we use in my company. So I would like to keep this behavior :-)
I have a proposal, you could add a new option like `--tr-force-testrun-creation` allowing to publish results in a new testrun and not in existing testruns.
Or (to avoid a new option) you could manage an additional value for option `--tr-run-id` like `new`:

 `--testrail --tr-config=testrail.cfg --tr-plan-id=70 --tr-run-id=new`

Reply:
The logic I patched to call the api and get a single new test run in the target plan [previously just assigned 0](https://github.com/allankp/pytest-testrail/blob/master/pytest_testrail/plugin.py#L156) as the run id before continuing as it does now, so now I'm curious what can/should happen if you do:

    --testrail --tr-config=testrail.cfg --tr-plan-id=70 --tr-run-id=0

How do you get the test runs into plans so they can be I guess wild card updated? Sounds strange, but we're working on our own test flow now, so who am I to critique. Actually, I'd really like to understand, since we need to figure out similar issues, and I'm not at all sure I want to always create a new run. (But probably sometimes, and inside a plan.)

I have no objection to preserving legacy functionality (as is); in fact I'd like to leverage all of it as much as I can. 

Do you have a need to keep the arguments as is for legacy? I'm asking because while I have no motive to change things from as is, passing in 0  for run id  to mean wildcard vs setting run id to a keyword like `new` , seems more intuitive to me and matches the original logic (FWIW). 

I'm also curious how we can/should use the run name, since that can already be passed in as well. From my experimentation if you pass in a run name but no run id and no plan id, you get a new run with the passed in name, over and over. I haven't tried using a plan id with a run name.

I think just adding a plan id should probably do something similar to calls without a plan id, create new runs, with or without a passed in name. When I got done experimenting with creating test runs with and without names outside of a test plan, then added just a test plan to the call, I got dramatically different functionality. With no test runs in place, nothing at all, and your telling me that if I did have runs in place, I'd have seen them all updated. My bug that got me to write this PR may be your feature.

    --testrail --tr-config=testrail.cfg --tr-run-name='New Run'

or

    --testrail --tr-config=testrail.cfg 

vs

    --testrail --tr-config=testrail.cfg --tr-plan-id=70 --tr-run-name='New Run?'

I'm not at all sure I'm happy with creating new runs all the time, we're likely to want to name runs soon, and I can see reasons to do what your doing, so I'm game for preserving the ah, feature.

Reply:
TL;DR I'll try to code it up as you suggest, conditional on a `new` token. I have questions about how it all is/should be working, but I'll try to make those clear in a commit to the README.md.

Reply:
New commits made including README.md. It will take a while to get appropriate unit tests added, but this is what I have in mind to enable more interaction with test plans.